lxc/list.go

```diff
@@ -281,16 +283,16 @@ func (c *listCmd) listContainers(d *lxd.Client, cinfos []shared.ContainerInfo, f
 	}
 
 	for _, cInfo := range cinfos {
-		cStatesLock.Lock()
-		cSnapshotsLock.Lock()
 		for _, column := range columns {
 			if column.NeedsState && cInfo.IsActive() {
 				_, ok := cStates[cInfo.Name]
 				if ok {
 					continue
 				}
 
+				cStatesLock.Lock()
 				cStates[cInfo.Name] = nil
+				cStatesLock.Unlock()
```

Comment:
These bottom two hunks are effectively a revert of afcffb844b0f4cbcf558800a636ee17ba51a3772, and enable concurrent reads again.